module3/homework/advanced_calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,81 @@
+#include "advancedCalculator.hpp"
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <functional>
+#include <map>
+#include <regex>
+
+double calculateFactorial(double number){
+    if(number <= 1){
+        return 1;
+    }
+    return std::tgamma(number + 1);
```

Comment:
```suggestion
    return number <= 1 ? 1 : std::tgamma(number + 1);
```
I suggest trying lamba expression here.